internal/storage/dummy_io_bucket_test.go

```diff
@@ -408,3 +469,188 @@ func TestDummyReader_ReadWithLatency(t *testing.T) {
 	assert.Equal(t, 512*1024, n)
 	assert.GreaterOrEqual(t, elapsed, 5*time.Millisecond)
 }
+
+////////////////////////////////////////////////////////////////////////
+// Test for dummyMultiRangeDownloader
+////////////////////////////////////////////////////////////////////////
+
+func TestDummyIOBucket_NewMultiRangeDownloader(t *testing.T) {
+	latency := 5 * time.Millisecond
+	params := DummyIOBucketParams{
+		PerMBLatency: latency,
+	}
+	dummyBucket := NewDummyIOBucket(&TestifyMockBucket{}, params)
+
+	mrd, err := dummyBucket.NewMultiRangeDownloader(context.Background(), &gcs.MultiRangeDownloaderRequest{})
+	dmrd, ok := mrd.(*dummyMultiRangeDownloader)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, mrd)
+	assert.True(t, ok)
+	assert.Equal(t, latency, dmrd.perMBLatency)
+}
+
+func TestDummyMultiRangeDownloader_Add_Single(t *testing.T) {
+	mrd := &dummyMultiRangeDownloader{}
+	var output bytes.Buffer
+	length := int64(100)
+	offset := int64(50)
+	var cbOffset, cbBytesWritten int64
+	var cbErr error
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	mrd.Add(&output, offset, length, func(o, bw int64, e error) {
+		cbOffset = o
+		cbBytesWritten = bw
+		cbErr = e
+		wg.Done()
+	})
+
+	wg.Wait() // Wait for callback to be called
```

Comment:
```suggestion
	wg.Wait() // Wait for callback to be called
```

nit: we can assume the wait as part of Act only (in AAA). Please make the change everywhere.